### PR TITLE
Al.ivanov

### PR DESCRIPTION
**Pull request by Inkln**
Centernet-based pipeline

### CODE REVIEW

File: src/callbacks.py [hunk 1]
```diff
@@ -5,15 +5,14 @@
 from .losses.ctdet_loss import decode_centernet_predictions
 from .metrics import class_agnostic_mean_ap, calculate_map, construct_mAP_list_from_bboxes
 
-from catalyst.dl import utils
-
 
 def detach(tensor) -> np.ndarray:
```
Comment (TezRomacH) [inline src/callbacks.py]: let's return `from catalyst.dl import utils` back and use `utils.detach` everywhere it needed

File: src/dataset.py [hunk 2]
```diff
@@ -54,9 +51,14 @@ def get_affine_transform(
     return trans
 
 
-def affine_transform(pt, t):
-    new_pt = np.array([pt[0], pt[1], 1.], dtype=np.float32).T
-    new_pt = np.dot(t, new_pt)
+def affine_transform(point: np.array, transform_matrix: np.array) -> np.array:
+    """
+    :param point: np.ndarray with shape (2, )
```
Comment (TezRomacH) [inline src/dataset.py]: we use [Google style docstring](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html)

File: src/experiment.py [hunk 3]
```diff
@@ -1,226 +1,58 @@
...
+        train_dataset = DetectionDataset(annotation_file=kwargs['train_annotation_file'],
+                                         images_dir=kwargs['train_images_dir'],
+                                         down_ratio=kwargs['down_ratio'],
+                                         max_objects=kwargs['max_objs'],
+                                         num_categories=kwargs['num_categories'],
+                                         image_size=kwargs['image_size'],
+                                         transform=train_transform(kwargs['image_size'][0])
+                                         )
+
+        # TODO TRAIN IS NOW EQUAL TO VAL
```
Comment (TezRomacH) [inline src/experiment.py]: why? 😄

File: src/experiment.py [hunk 4]
```diff
@@ -1,226 +1,58 @@
...
-    for (f, b, l) in samples_:
-        filepaths.append(f)
-        bboxes.append(b)
-        labels.append(l)
-
-    tqdm.write(f"{samples_path}: {len(filepaths), len(bboxes), len(labels)}")
-    return filepaths, bboxes, labels, ages
-
-
 class Experiment(ConfigExperiment):
```
Comment (TezRomacH) [inline src/experiment.py:13]: thanks, for cleaning up all the `_load_data`, `run_one` methods ❤️

File: src/runner.py [hunk 5]
```diff
@@ -6,10 +6,12 @@ def __init__(
         self,
         model=None,
         device=None,
+        input_key=None,
```
Comment (TezRomacH) [inline src/runner.py]: so, since you pass `runner_args` to the config, you can easily remove this custom `Runner` class

and in `__init__.py` use `from catalyst.dl import SupervisedRunner as Runner`. That's all 🙂

**Review by TezRomacH — CHANGES_REQUESTED** [review]

File: src/callbacks.py (hunk 1 above)
Comment (Inkln) [reply src/callbacks.py]: done

File: src/experiment.py (hunk 3 above)
Comment (Inkln) [reply src/experiment.py]: Because current config file example shows: 
```
   # default values, will be used if something aren't specified
    annotation_file: ./data/annotation.json
    images_dir: ./data/annotation.json

    # You may specify next parameters, data source will be overwritten
    #train_annotation_file: ./data_train/annotation.json
    #valid_annotation_file: ./data_valid/annotation.json
    #train_images_dir: ./data_train/images/
    #valid_images_dir: ./data_valid/images/
```
I'am able to fix it here

File: src/runner.py (hunk 5 above)
Comment (Inkln) [reply src/runner.py]: done

**Comment by Inkln** [comment]
@TezRomacH  Is it all?)

**Comment by TezRomacH** [comment]
Yeah, thanks for pinging me :)

Can you please add a README describing how the dataset format is organized, how to launch the training and where will the results lie?

Why did you rename `num_classes` to `num_categories`? In all other pipelines, we use `num_classes`.

Also, in some parts there are comments in the format `:params:/:return:`, and you should use Google Style.

File: requirements.txt [hunk 6]
```diff
@@ -1,3 +1,6 @@
 catalyst
```
Comment (Scitator) [inline requirements.txt]: could you please specify catalyst version? for 20.1 for example

Comment (Inkln) [reply requirements.txt]: Done

**Review by Inkln — COMMENTED** [review]
> Yeah, thanks for pinging me :)
> 
> Can you please add a README describing how the dataset format is organized, how to launch the training and where will the results lie?
> 
> Why did you rename `num_classes` to `num_categories`? In all other pipelines, we use `num_classes`.
> 
> Also, in some parts there are comments in the format `:params:/:return:`, and you should use Google Style.

Done

**Comment by Scitator** [comment]
@Avi2011class So, is this pipeline ready for our alpha test? How do you think?

**Comment by Inkln** [comment]
> @Avi2011class So, is this pipeline ready for our alpha test? How do you think?

Yes.
